### PR TITLE
NZ 22 update map ui components

### PR DESCRIPTION
**Pull request by tcaiger**
### Changes:

- Update UI Components Map Components with functionality for web frontend
- Update LESMIS to use new UI Component Map Components

---

### CODE REVIEW

File: packages/ui-components/src/components/Map/InteractivePolygon.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+`;
+
+const parseProps = (organisationUnitCode, organisationUnitChildren, measureOrgUnits) => {
+  let shade;
+  let isHidden;
+  let orgUnitMeasureData;
+  let hasShadedChildren = false;
+
+  if (measureOrgUnits.length > 0) {
+    const measureOrgUnitCodes = measureOrgUnits.map(orgUnit => orgUnit.organisationUnitCode);
```
Comment (edmofro) [inline packages/ui-components/src/components/Map/InteractivePolygon.js:44]: A simple change to make this code block more efficient would be to use a Set here and a `.has` hashed lookup wherever `.includes` is currently running a full loop across the array
```suggestion
    const measureOrgUnitCodes = new Set(measureOrgUnits.map(orgUnit => orgUnit.organisationUnitCode));
```

File: packages/ui-components/src/components/Map/LeafletMap.js [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+    return arePositionsEqual(a[0], b[0]) && arePositionsEqual(a[1], b[1]);
+  }
+  // if either are not defined, equal only if both are not defined
+  return a === b;
+}
+
+function areBoundsValid(b) {
+  return Array.isArray(b) && b.length === 2;
+}
+export class LeafletMap extends Component {
```
Comment (edmofro) [inline packages/ui-components/src/components/Map/LeafletMap.js:50]: Would have thought there'd be a prettier rule about this, but should there be a space above the class definition?

File: packages/ui-components/src/components/Map/LeafletMap.js [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+    this.lat = center.lat;
+    this.lng = center.lng;
+
+    this.moving = false;
+
+    this.onPositionChanged();
+  };
+
+  onMoveStart = () => {
+    this.moving = false;
```
Comment (edmofro) [inline packages/ui-components/src/components/Map/LeafletMap.js:165]: This looks the wrong way around to me

**Review by edmofro — CHANGES_REQUESTED** [review]
Hard to keep my head across all of this! Have focused more on details than the bigger picture, perhaps we should have a walk and talk across all four for bigger picture stuff

File: packages/ui-components/src/components/Map/LeafletMap.js (hunk 2 above)
Comment (tcaiger) [reply packages/ui-components/src/components/Map/LeafletMap.js:50]: Yes agreed. No there's no rule around it though.

File: packages/ui-components/src/components/Map/LeafletMap.js (hunk 3 above)
Comment (tcaiger) [reply packages/ui-components/src/components/Map/LeafletMap.js:165]: Yes good catch

File: packages/ui-components/src/components/Map/InteractivePolygon.js (hunk 1 above)
Comment (tcaiger) [reply packages/ui-components/src/components/Map/InteractivePolygon.js]: Great suggestion. I will start using Sets more

File: packages/ui-components/src/components/Map/MarkerLayer.js [hunk 4]
```diff
@@ -23,27 +22,36 @@ const ShadedPolygon = styled(Polygon)`
...
-      {measureData.map(measure =>
+      {data.map((measure, index) =>
         measure.region ? (
-          <ShadedPolygon
-            key={`${measure.code}-${measure.color}`}
-            positions={measure.region}
-            {...measure}
-          >
+          // eslint-disable-next-line react/no-array-index-key
+          <ShadedPolygon key={index} positions={measure.region} {...measure}>
```
[the remark this answers is not in the document]
Comment (tcaiger) [reply packages/ui-components/src/components/Map/MarkerLayer.js:49]: Yes good point! I agree!

**Comment by tcaiger** [comment]
@edmofro - This is ready for re-review. Maybe we can merge the 2 PRs and then do a walk and talk on the final PR. Also the fix for tests failing is on the other branch.

**Review by edmofro — APPROVED** [review]
Yep that strategy (merge then walk and talk) sounds good to me :-)

**Comment by tcaiger** [comment]
@edmofro - Are you happy for this to go to test?

**Comment by edmofro** [comment]
Yes!

**Pull request merged**